### PR TITLE
model: add APMEvent.{Observer,ECSVersion}

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Introduce Observer and ECSVersion to APMEvent, and populate them with model processors. This replaces the libbeat processors that were setting these fields.

Add some missing fields to the mappings, which became evident when switching to strict mapping in the integration.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
~- [ ] Documentation has been updated~

## How to test these changes

Non-functional change.

## Related issues

None

### CODE REVIEW

File: beater/processors.go [hunk 1]
```diff
@@ -57,3 +61,37 @@ func rateLimitBatchProcessor(ctx context.Context, batch *model.Batch) error {
...
+	for i := range *b {
+		event := &(*b)[i]
+		event.ECSVersion = ecs.Version
+	}
+	return nil
+}
+
+// newBeatInfoBatchProcessor returns a model.BatchProcessor that sets observer
+// fields from info.
+func newBeatInfoBatchProcessor(info beat.Info) model.ProcessBatchFunc {
```
Comment (simitt) [inline beater/processors.go:77]: How about calling this `observerBatchProcessor`?

**Review by simitt — APPROVED** [review]
Just for my understanding - the additional definitions for indexed fields are required for the apmpackage as fields from libbeat templates are not inherited there right?

**Comment by axw** [comment]
> Just for my understanding - the additional definitions for indexed fields are required for the apmpackage as fields from libbeat templates are not inherited there right?

As discussed live: yes.

File: beater/processors.go (hunk 1 above)
Comment (axw) [reply beater/processors.go]: Yes, that is a better name. Done.

**Pull request merged**